File: thinking-categorically/category/01-ideas/ideas.ts
const compose = <A,B,C>(g:(_:B)=> C, f:(_:A)=> B) =>
    (x:A) => g(f(x));

const f = (x:number) => x.toString();      // Number -> String
const g = (s:string) => s.length > 2;      // String -> Boolean

const h = compose(g, f);          // Number -> Boolean

console.log(h(123)); // Output: true

File: thinking-categorically/category/01-ideas/ideas.cs
Func<A, C> Compose<A, B, C>(Func<B, C> g, Func<A, B> f)
    => x => g(f(x));

Func<int, string> f = x => x.ToString();
Func<string, bool> g = s => s.Length > 2;

var h = Compose(g, f); // Func<int, bool>
Console.WriteLine(h(123)); // Output: True
